Swift/Collection.swift

```diff
@@ -0,0 +1,292 @@
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// A `Collection` represent a collection which is a container for documents.
+///
+/// A collection can be thought as a table in the relational database. Each collection belongs to
+/// a scope which is simply a namespce, and has a name which is unique within its scope.
+///
+/// When a new database is created, a default collection named `_default` will be automatically
+/// created. The default collection is created under the default scope named `_default`.
+/// You may decide to delete the default collection, but noted that the default collection cannot
+/// be re-created. The name of the default collection and scope can be referenced by using
+/// `Collection.defaultCollectionName` and `Scope.defaultScopeName` constant.
+///
+/// When creating a new collection, the collection name, and the scope name are required.
+/// The naming rules of the collections and scopes are as follows:
+/// - Must be between 1 and 251 characters in length.
+/// - Can only contain the characters A-Z, a-z, 0-9, and the symbols _, -, and %.
+/// - Cannot start with _ or %.
+/// - Both scope and collection names are case sensitive.
+///
+/// ## CBLCollection Lifespan
+/// A `Collection` object and its reference remain valid until either
+/// the database is closed or the collection itself is deleted, in that case it will
+/// return CBLErrorNotOpen while accessing the collection APIs.
+///
+/// ## Legacy Database and API
+/// When using the legacy database, the existing documents and indexes in the database will be
+/// automatically migrated to the default collection.
+///
+/// Any pre-existing database functions that refer to documents, listeners, and indexes without
+/// specifying a collection such as `database.document(id:)` will implicitly operate on
+/// the default collection. In other words, they behave exactly the way they used to, but
+/// collection-aware code should avoid them and use the new Collection API instead.
+/// These legacy functions are deprecated and will be removed eventually.
+///
+public final class Collection : CollectionChangeObservable, Indexable {
+    
+    /// The default scope name constant
+    public static let defaultCollectionName: String = "_default"
+    
+    /// Collection name.
+    var name: String { _impl.name }
+    
+    /// The scope of the collection.
+    var scope: Scope { Scope(_impl.scope) }
+    
+    // MARK: Document Management
+    
+    /// Total number of documents in the collection.
+    var count: UInt64 { _impl.count }
+    
+    /// Get an existing document by id.
+    ///
+    /// CBLErrorNotOpen code will be thrown if the collection is deleted
```

Comment:
CBLErrorNotOpen > An error with CBLError.notOpen code or `CBLError.notOpen code`.

Similarly for the other methods as well.